packages/data/src/components/with-dispatch/index.js

```diff
@@ -54,8 +54,11 @@ const withDispatch = ( mapDispatchToProps ) => createHigherOrderComponent(
 					// called, it is done only to determine the keys for which
 					// proxy functions should be created. The actual registry
 					// dispatch does not occur until the function is called.
-					const propsToDispatchers = mapDispatchToProps( this.props.registry.dispatch, props.ownProps );
+					const propsToDispatchers = mapDispatchToProps( this.props.registry.dispatch, props.ownProps, this.props.registry.select );
 					this.proxyProps = mapValues( propsToDispatchers, ( dispatcher, propName ) => {
+						if ( typeof dispatcher !== 'function' ) {
+							throw new TypeError( `${ propName } returned from mapDispatchToProps in withDispatch HOC must be a function` );
```

Comment:
I'm wondering whether we should only `console.warn` here instead. If any plugin uses non-function here it will explode the editor :(

Reply:
In some sense I like the explosion for making the issue obvious 😆 

A warn should be enough to be flagged in our end-to-end tests, assuming the specific functionality is covered.

Reply:
Let's start with `console.warn`. It looks like using a non-function property to render will explode component anyways. We are covered here. However, there still might be plugins which pass non-function properties to use with callbacks. I don't want to take a risk of breaking them just before 5.0 release. A warning should be good for start.